# HG changeset patch
# User Rik <rdrider0-list@yahoo.com>
# Date 1260772367 28800
#      Sun Dec 13 22:32:47 2009 -0800
# Node ID e352f8366b02f1545afafc551b5ea148c91f4f2f
# Parent  3878bae57c12d0e026bbdc57cd24f35194800078
Use simple_move_if_change_rule for cruft.def to eliminate unnecessary compilations.

diff --git a/libcruft/ChangeLog b/libcruft/ChangeLog
--- a/libcruft/ChangeLog
+++ b/libcruft/ChangeLog
@@ -1,8 +1,13 @@
+2009-12-13  Rik <octave@nomad.inbox5.com>
+
+	* Makefile.am: Use simple_move_if_change_rule for cruft.def to eliminate
+	unnecessary compilations.
+
 2009-12-07  John W. Eaton  <jwe@octave.org>
 
 	* misc/f77-fcn.h, misc/lo-error.h, misc/quit.h,
 	misc/f77-extern.cc, misc/quit.cc, misc/cquit.c, misc/f77-fcn.c,
 	misc/lo-error.c: Remove Emacs local variable list.
 
 2009-12-07  John W. Eaton  <jwe@octave.org>
 
diff --git a/libcruft/Makefile.am b/libcruft/Makefile.am
--- a/libcruft/Makefile.am
+++ b/libcruft/Makefile.am
@@ -18,19 +18,19 @@
 # You should have received a copy of the GNU General Public License
 # along with Octave; see the file COPYING.  If not, see
 # <http://www.gnu.org/licenses/>.
 
 TOPDIR = ..
 
 include ../common.mk
 
-octlib_LTLIBRARIES = libcruft.la
+AUTOMAKE_OPTIONS = subdir-objects
 
-AUTOMAKE_OPTIONS = subdir-objects
+octlib_LTLIBRARIES = libcruft.la
 
 libcruft_la_SOURCES =
 
 libcruft_la_CPPFLAGS = @CRUFT_DLL_DEFS@
 
 libcruft_la_LDFLAGS = \
   -release $(version) $(NO_UNDEFINED_LDFLAG) @XTRA_CRUFT_SH_LDFLAGS@
 
@@ -55,12 +55,13 @@ include misc/module.mk
 include odepack/module.mk
 include ordered-qz/module.mk
 include quadpack/module.mk
 include ranlib/module.mk
 include slatec-err/module.mk
 include slatec-fn/module.mk
 include villad/module.mk
 
-cruft.def: $(libcruft_la_SOURCES) $(TOPDIR)/libcruft/mkf77def
+cruft.def: $(libcruft_la_SOURCES) mkf77def
 	chmod a+rx mkf77def
 	./mkf77def $(srcdir) $(libcruft_la_SOURCES) > $@-t
-	mv $@-t $@
+	$(simple_move_if_change_rule)
+
